mesheryctl/internal/cli/root/config/config.go

```diff
@@ -53,13 +54,43 @@ func (mc *MesheryCtlConfig) CheckIfCurrentContextIsValid() (Context, error) {
 	if mc.CurrentContext == "" {
 		return Context{}, errors.New("current context not set")
 	}
+
 	if ctx, exists := mc.Contexts[mc.CurrentContext]; exists {
-		if ctx.Version == "" {
-			ctx.Version = "latest"
+		err := ctx.ValidateVersion()
+
+		if err != nil {
+			return Context{}, errors.New("invalid version " + ctx.Version + " specified")
+		}
+
+		if err == nil {
+			return ctx, nil
 		}
-		return ctx, nil
 	}
-	return Context{}, errors.New("current context:" + mc.CurrentContext + " does not exist")
+
+	return Context{}, errors.New("current context " + mc.CurrentContext + " does not exist")
+}
+
+// ValidateVersion checks if the version is valid, if empty sets it to default value latest. Returns an error if the version is invalid.
+func (ctx *Context) ValidateVersion() error {
+	if ctx.Version == "" {
+		ctx.Version = "latest"
+		return nil
+	}
+
+	if ctx.Version == "latest" {
+		return nil
+	}
+
+	matched, err := regexp.MatchString(`v[0-9]+.[0-9]+.[0-9]+[-a-z0-9]*`, ctx.Version)
+	if err != nil || !matched {
+		return errors.New("invalid version " + ctx.Version + " specified")
+	}
+
+	if matched {
+		return nil
+	}
+
+	return errors.New("invalid version " + ctx.Version + " specified")
```

Comment:
This errors are propagated to where the `ValidateVersion` is called right? Which is line 59. So in line 62 we can just pass the err instead of a new error. This would not change any output. But it would be cleaner.

Reply:
@navendu-pottekkat understood!
Just check these changes.
Thanks!